mqsf/service.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) 2023 SUSE LLC.  All rights reserved.
+#
+# This file is part of mqsf.
+#
+
+import logging
+
+from amqpstorm import Connection
+
+# project
+from mqsf.log.filter import BaseServiceFilter
+from mqsf.exceptions import MQConnectionException
+from mqsf.utils import setup_mq_log_handler
+
+
+class Service(object):
+    """
+    Base class for RabbitMQ message broker
+
+    Attributes
+
+    * :attr:`host`
+      RabbitMQ server host
+
+    * :attr:`service_exchange`
+      Name of service exchange
+    """
+    def __init__(self, service_exchange, config, custom_args=None):
+        self.channel = None
+        self.connection = None
+
+        self.service_exchange = service_exchange
+        self.custom_args = custom_args
+        self.config = config
+
+        # amqp settings
+        self.amqp_host = self.config.get_amqp_host()
+        self.amqp_user = self.config.get_amqp_user()
+        self.amqp_pass = self.config.get_amqp_pass()
+
+        self._open_connection()
+
+        logging.basicConfig()
+        self.log = logging.getLogger(
+            '{0}Service'.format(self.service_exchange.title())
+        )
+        self.log.setLevel(logging.DEBUG)
+        self.log.propagate = False
+
+        mq_handler = setup_mq_log_handler(
+            self.amqp_host,
+            self.amqp_user,
+            self.amqp_pass
+        )
+        self.log.addHandler(mq_handler)
+        self.log.addFilter(BaseServiceFilter())
+
+        self.post_init()
+
+    def post_init(self):
+        """
+        Post initialization method
+
+        Implementation in specialized service class
+        """
+        pass
+
+    def _declare_direct_exchange(self, exchange):
+        """
+        Declare/create exchange and set as durable.
+
+        The exchange, queues and messages will survive a broker restart.
+        """
+        self.channel.exchange.declare(
+            exchange=exchange, exchange_type='direct', durable=True
+        )
+
+    def _declare_queue(self, queue):
+        """
+        Declare the queue and set as durable.
+        """
+        return self.channel.queue.declare(queue=queue, durable=True)
+
+    def _get_queue_name(self, exchange, name):
+        """
+        Return formatted name based on exchange and queue name.
+
+        Example: obs.service
+        """
+        return '{0}.{1}'.format(exchange, name)
+
+    def _open_connection(self):
+        """
+        Open connection or channel if currently closed or None.
+
+        Raises: MashRabbitConnectionException if connection
+                cannot be established.
+        """
+        if not self.connection or self.connection.is_closed:
+            try:
+                self.connection = Connection(
+                    self.amqp_host,
+                    self.amqp_user,
+                    self.amqp_pass,
+                    kwargs={'heartbeat': 600}
+                )
+            except Exception as e:
+                raise MashRabbitConnectionException(
```

Comment:
Should be MQConnectionException

Reply:
Also, found a couple other references to RabbitMQ